src/regdesk/printer_manager.js

```diff
@@ -0,0 +1,66 @@
+import { LabelEpl, LP2844 } from 'WebZLP/src/LP2844'
+
+export class PrinterManager {
+    #devices = [];
+    #printers = [];
+
+    primaryPrinter = 0;
+    secondaryPrinter = 1;
+
+    #printerType = LP2844;
+
+    constructor() {
+    }
+
+    get printers() { return this.#printers; }
+
+    async printerConnect(e) {
+        if (!this.#devices.includes(e.device)) {
+            this.#devices.push(e.device);
+            let printer = new this.#printerType(e.device);
+            await printer.connect();
+            this.#printers.push(printer);
+            this.primaryPrinter = 0;
+        }
+    }
+
+    async printerDisconnect(e) {
+        const idx = this.#devices.findIndex(i => e.device == i);
+        if (idx > -1) {
+            const p = this.#printers[idx];
+            this.#devices.splice(idx, 1);
+            this.#printers.splice(idx, 1);
+            this.resetPrinterButtons();
+            await p.dispose();
+        }
+    }
+
+    async printerPair() {
+        try {
+            let device = await navigator.usb.requestDevice({
+                filters: [
+                    { vendorId: this.#printerType.usbVendorId }
+                ]
+            });
+
+            await this.printerConnect( { device : device} )
+        } catch (e) {
+            console.log("Failed to connect to printer!");
+            console.log(e);
+            return;
```

Comment:
Should this return false or something? I assume that the client needs to know that pair failed?

Reply:
Yes, and everything should probably return something more useful than nothing. I'll do a cleanup pass to make sure the interface surface is a little more coherent.